### PR TITLE
HTCONDOR-2493 unprivileged cgroup v2

### PR DESCRIPTION
**Pull request by GregThain**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/proc_family_direct_cgroup_v2.cpp [hunk 1]
```diff
@@ -237,6 +237,8 @@ ProcFamilyDirectCgroupV2::cgroupify_process(const std::string &cgroup_name, pid_
 			dprintf(D_ALWAYS, "Error writing procid %d to %s: %s\n", pid, procs_filename.c_str(), strerror(errno));
 			close(fd);
 			return false;
+		} else {
+			dprintf(D_ALWAYS, "Successfully moved procid %d to cgroup %s\n", pid, procs_filename.c_str());
```
Comment (Todd-L-Miller) [inline src/condor_utils/proc_family_direct_cgroup_v2.cpp:241]: D_ALWAYS for a successful operation?  Should probably become something else before this knob becomes the default.

**Review by Todd-L-Miller — APPROVED** [review]
Looks good to me.

**Comment by Todd-L-Miller** [comment]
There's probably infrastructural work to do here, but this seems like a big-enough win that we very much want to have a regression test for it.  Is the code in a state where we can do that?  (It seems like, but I don't know what else might be required.)  The infrastructural work may merit its own ticket?

**Comment by GregThain** [comment]
> There's probably infrastructural work to do here, but this seems like a big-enough win that we very much want to have a regression test for it. Is the code in a state where we can do that? (It seems like, but I don't know what else might be required.) The infrastructural work may merit its own ticket?

Totally agree.  I can run the test suite "by hand" with this option set, but I'd like to run all batlab runs with this on.  To do so requires the docker universe batlab jobs to give their containers a writeable cgroup.  It seems like we should be able to do so by having the C1 condor launch docker universe jobs with the docker command line `--cgroup-parent`, but I cannot get this to work by hand on the docker run command line.

**Pull request merged**